src/Pool.sol

```diff
@@ -755,50 +657,46 @@ abstract contract Pool {
         }
         sender.startBalance -= (_currTimestamp() - sender.startTime) * amtPerSec;
         // Set negative deltas to clear deltas applied by the previous call to `_startSending`
-        _setDeltasFromNow(weights, -int128(amtPerWeight), endTime, updates);
+        _setDeltasFromNow(receivers, -1, endTime, updates);
     }
 
     /// @notice Makes the user start sending funds.
     /// It applies effects of the sender on all of its receivers.
     /// It doesn't modify the sender.
     /// @param sender The updated sender
-    /// @param weights The list of the user's receivers and their weights.
+    /// @param receivers The list of the user's receivers.
     /// @param updates The list of stream updates to log
     function _startSending(
         Sender storage sender,
-        ReceiverWeight[] calldata weights,
+        Receiver[] calldata receivers,
         StreamUpdates memory updates
     ) internal {
-        uint32 weightsSum = _weightsSum(weights);
+        uint128 amtPerSec = _totalAmtPerSec(receivers);
         // Won't be sending anything
-        if (weightsSum == 0 || sender.amtPerSec < weightsSum) return;
-        uint128 amtPerWeight = sender.amtPerSec / weightsSum;
-        uint128 amtPerSec = amtPerWeight * weightsSum;
+        if (amtPerSec == 0) return;
         // Won't be sending anything
         if (sender.startBalance < amtPerSec) return;
         sender.startTime = _currTimestamp();
         uint256 endTimeUncapped = _currTimestamp() + uint256(sender.startBalance / amtPerSec);
         uint64 endTime = endTimeUncapped > MAX_TIMESTAMP ? MAX_TIMESTAMP : uint64(endTimeUncapped);
-        _setDeltasFromNow(weights, int128(amtPerWeight), endTime, updates);
+        _setDeltasFromNow(receivers, 1, endTime, updates);
```

Comment:
Maybe we can make a constant for the negative and positive multiplier and pass it here?

Reply:
The multiplier is a quick and dirty solution, the whole `_setDeltasFromNow` will be removed in the next PR

Reply:
okay. perfecto